src/screens/PasswordRecovery/componentes/EmailForm/index.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import {
+  KeyboardAvoidingView,
+  TextInput
+} from 'react-native'
+import * as Yup from 'yup'
+import { Formik } from 'formik'
+import { isEmpty, values } from 'lodash'
+import { Body, Description, GrayButton, PrimaryButton } from '~/components'
+import styles, { PLACEHOLDER_COLOR } from './styles'
+
+const initialValues = {
+  email: ''
+}
+
+const Error = ({ error }) => (
+  error && (
+    <Body style={styles.error}>
+      Email invalido
+    </Body>
+  )
+)
+
+const schema = Yup.object().shape({
+  email: Yup.string()
+    .email('Email inválido')
+    .required('Requerido')
+})
+
+const formHasBlankValues = (formValues) => {
+  const blankValues = values(formValues).filter(element => element === '')
+  return !isEmpty(blankValues)
+}
+
+const SubmitButton = ({ onPress, blankValuesExist, error, disabled, children }) => {
+  if (blankValuesExist || error) {
+    return (
+      <GrayButton style={styles.submitButton}>
+        <Description white>
+          {children}
+        </Description>
+      </GrayButton>
+    )
+  } else {
+    return (
+      <PrimaryButton
+        onPress={onPress}
+        style={styles.submitButton}
+        disabled={disabled}
+      >
+        <Description white>
+          {children}
+        </Description>
+      </PrimaryButton>
+    )
+  }
+}
+
+const EmailForm = ({
+  handleSubmit,
+  error,
+  applyValidation
+}) => {
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={applyValidation && schema}
+    >
+      {formikProps => (
+        <KeyboardAvoidingView style={styles.container}>
+          <TextInput
+            style={styles.label}
+            placeholder='Ingresá tu e-mail'
+            placeholderTextColor={PLACEHOLDER_COLOR}
+            onChangeText={formikProps.handleChange('email')}
+            onBlur={formikProps.handleBlur('email')}
+            value={formikProps.values.email}
+          />
+          {formikProps.errors.email && formikProps.touched.email &&
+            <Body style={styles.errorMessage}>{formikProps.errors.email}</Body>}
+          <Error error={error} />
+          <SubmitButton
+            onPress={formikProps.handleSubmit}
+            blankValuesExist={formHasBlankValues(formikProps.values)}
+            error={!formikProps.isValid}
+            disabled={formHasBlankValues(formikProps.values)}
```

Comment:
Why not use `disabled` only? It seems redundant, these three props all do the same: disable the button.